docs/Dealing-with-Duplication-of-Data.md

```diff
@@ -0,0 +1,108 @@
+* There is no well-defined order how configuration is applied (in some cases the file overrides the manifest, in others it doesn't, in other cases information is not applies at all)
+
+## Abstraction levels of configuration
+
+The underlying issue is that configuration can be represented on different levels of abstraction:
+
+### Raw data
+
+Looking at configuration files as opaque objects preserves all the state, but doesn't provide insight into the actual configuration. This would be the represenation as raw data without any processing.
+
+### Parsed raw data
+
+If the format of a configuration file is known, it can be parsed into a more high-level representation maintaining all information in a generic way (e.g. parsing an INI-style config file into a JSON representation). This can provide some generic insight into the configuration, but requires the user to still have knowledge about the details of the configuration and how its represented. There also is a chance that some minor information is lost, when the parser is not perfect (e.g. comments, ordering of entries, etc.)
+
+### Extracted information
+
+For known configuration it's possible to explicitly extract the relevant configuration information from files and store it in a more general abstracted format in the manifest. This gives a high degree of insight, possibly across different ways and tools to manage the configuration. It is hard to make sure that all information is extracted, though. Especially if configuration changes, even a complete extraction might become inconsistent, without the user noticing it.
+
+### Declarative configuration
+
+The highest level of abstraction would be to have a declaration of intended result instead of concrete configuration settings. For example a service can be described like a black box (e.g. webserver listening on port 80) or in detail (e.g. apache's configuration files). This declarative configuration would be on the level which typically is handled by configuration management systems (such as Puppet or SaltStack). It might be hard or even impossible to extract information on this level from a system as it can require information about the intent of configuration, which might not be stored in the system itself. It often would be the desired level of abstraction when defining the configuration of a system. Typically it would be stored outside of the systems itself.
+
+
+## Processing configuration
+
+To get higher levels of abstraction it is necessary to process configuration files and data.
+
+One challenge is to correctly process configuration. In some cases the only reliable way how to do that is to use the native tools using the configuration. As a middle ground existing abstractions of parsers, such as Augeas or what is in YaST could be helpful.
```

Comment:
As a side note: An idea how to get an abstraction of data might be 'highjacking' chef or puppet. These tools have a lot of knowledge about services and their configuration. We could try to make use of chef or puppet as a library for parsing the configuration and getting an abstraction of data. That's for sure not trivial. It might be an exciting research project, though.